src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -165,6 +165,13 @@ public function validateProperty(object $object, string $propertyName, string|Gr
             throw new ValidatorException(\sprintf('The metadata factory should return instances of "\Symfony\Component\Validator\Mapping\ClassMetadataInterface", got: "%s".', get_debug_type($classMetadata)));
         }
 
+        if (!$classMetadata->hasPropertyMetadata($propertyName)) {
+            $className = \is_object($object) ? $object::class : $object;
+            if (!property_exists($className, $propertyName) && !method_exists($className, 'get'.ucfirst($propertyName)) && !method_exists($className, 'is'.ucfirst($propertyName)) && !method_exists($className, 'has'.ucfirst($propertyName))) {
```

Comment:
this doesn't look appropriate to me: these checks rely on knowledge about what's a property that's none of this class' business